### DISCUSSION TITLE
Channel message log search with list of IDs

### DISCUSSION THREAD

**Discussion by ropese34** [nextgenhealthcare/connect#5544] [Feature/Enhancement Ideas]
Hello, I want to pull up messages referencing a list of these particular patient IDs. What is the best way to perform a single search with for this list of IDs? Thanks

**Comment by ropese34** [comment]
@pacmano1

**Comment by jonbartels** [comment]
How large of a date range or message count are you searching over?

The BEST way to do this is to change your channel so that the patient ID is parsed to a connectorMap value, then add that value to metadata. https://docs.nextgen.com/bundle/Mirth_User_Guide_41/page/connect/connect/topics/t_Metadata_Table_connect_ug.html . Then you can just do a metadata search.

The best approach doesn't work for messages prior to when that info was added. 

The next option is to do an advanced search and search specifically in maps or a single message type for a `contains` with your patient ID *AND A DATE RANGE CONSTRAINT*. This speeds the search by looking in only the raw message (for example) and over a smaller set of messages (by date).

**Comment by ropese34** [comment]
> How large of a date range or message count are you searching over?
> 
> The BEST way to do this is to change your channel so that the patient ID is parsed to a connectorMap value, then add that value to metadata. https://docs.nextgen.com/bundle/Mirth_User_Guide_41/page/connect/connect/topics/t_Metadata_Table_connect_ug.html . Then you can just do a metadata search.
> 
> The best approach doesn't work for messages prior to when that info was added.
> 
> The next option is to do an advanced search and search specifically in maps or a single message type for a `contains` with your patient ID _AND A DATE RANGE CONSTRAINT_. This speeds the search by looking in only the raw message (for example) and over a smaller set of messages (by date).

Thank you. I wanted to perform a single search of the raw/source messages for multiple patient IDs. We have the basic open source version.

**Reply by pacmano1** [reply]
In the text search use a regex, simple example below.

e.g. `(MRN20624306|MRN234587)`.
